File: Java/Examples/ListViewDemo/app/src/main/java/com/example/listviewdemo/MainActivity.java
package com.example.listviewdemo;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;

/** @Autor Created by Anubys on the 12.03.2020 */


public class MainActivity extends AppCompatActivity {
    private static final String TAG = MainActivity.class.getSimpleName();

    @Nullable
    @BindView(R.id.lv_view)
    ListView listView;

    private Unbinder unbinder;
    private ArrayList<String> friends;


    //* ************************************************ *
    //*               L I F E - C Y C L E                *
    //* ************************************************ *
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        Log.d(TAG, "TAG - MainActivity - onCreate()");
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        unbinder = ButterKnife.bind(this);

        init();
        setListener();
    }

    @Override
    protected void onDestroy() {
        Log.d(TAG, "TAG - MainActivity - onDestroy()");
        super.onDestroy();

        if (unbinder != null) {
            unbinder.unbind();
        }
    }


    //* ************************************************ *
    //*         H E L P E R  -  M E T H O D S            *
    //* ************************************************ *
    private void init() {
        Log.d(TAG, "TAG - MainActivity - init()");
        friends = new ArrayList<>();
        friends.add("Boris");
        friends.add("Christine");
        friends.add("Jens");
        friends.add("Edward");
        friends.add("Martin");
        friends.add("Phillip");
        friends.add("Florian");

        ArrayAdapter<String> arrayAdapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, friends);

        if (listView != null) {
            listView.setAdapter(arrayAdapter);
        }
    }

    private void setListener() {
        Log.d(TAG, "TAG - MainActivity - setListener()");

        if (listView != null) {
            listView.setOnItemClickListener(new onItemClickListener());
        }
    }


    //* ************************************************ *
    //*                  C L A S S E S                   *
    //* ************************************************ *
    private class onItemClickListener implements android.widget.AdapterView.OnItemClickListener {

        @Override
        public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
            Toast.makeText(getApplicationContext(), getResources().getString(R.string.txt_hello) + " " + friends.get(position), Toast.LENGTH_LONG).show();
        }
    }
}

File: Kotlin/Examples/CustomListViewWithCardview/app/src/main/java/com/anubys/example/customlistviewwithcardview/MainActivity.kt
package com.anubys.example.customlistviewwithcardview

/** @Author Created by Anubys on the 11.08.2020 */

import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.AdapterView
import android.widget.Toast

import androidx.appcompat.app.AppCompatActivity

import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity(), AdapterView.OnItemClickListener {
    private val tag = MainActivity::class.java.simpleName

    private var itemAdapters: ItemAdapters? = null
    private var arrayList: ArrayList<ItemList>? = null


    //* ************************************************ *
    //*               L I F E - C Y C L E                *
    //* ************************************************ *
    override fun onCreate(savedInstanceState: Bundle?) {
        Log.v(tag,"TAG - MainActivity - onCreate()")
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        init()
    }

    override fun onItemClick(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
        Log.v(tag,"TAG - MainActivity - onItemClick()")
        val items: ItemList? = arrayList?.get(position)
        Toast.makeText(applicationContext, items?.title, Toast.LENGTH_LONG).show()
    }


    //* ************************************************ *
    //*         H E L P E R  -  M E T H O D S            *
    //* ************************************************ *
    private fun init() {
        Log.v(tag,"TAG - MainActivity - init()")
        arrayList = ArrayList()
        arrayList = setDataItem()
        itemAdapters = ItemAdapters(applicationContext, arrayList!!)
        listview?.adapter = itemAdapters
        listview?.onItemClickListener = this
    }

    private fun setDataItem(): ArrayList<ItemList> {
        Log.v(tag,"TAG - MainActivity - setDataItem()")
        val listItem: ArrayList<ItemList> = ArrayList()
        listItem.add(ItemList(R.drawable.ic_launcher_background, "C Language", "C is a ....."))
        listItem.add(ItemList(R.drawable.ic_launcher_background, "C++ Language", "C++ is a ....."))
        listItem.add(ItemList(R.drawable.ic_launcher_background, "Java Language", "Java is a ....."))
        listItem.add(ItemList(R.drawable.ic_launcher_background, "Kotlin Language", "Kotlin is a ....."))

        return (listItem)
    }

}
